portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,51 +29,45 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.Arrays;
-import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    ArrayList<String> words = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
-    
+    //query for comment storage through datastore.
+    Query query = new Query("Task");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
-    //iterating over entities within Datastore
+    //populate comments ArrayList with entities within Datastore.
+    ArrayList<String> comments = new ArrayList<String>();
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
       String title = (String) entity.getProperty("user-comments");
-      long timestamp = (long) entity.getProperty("timestamp");
+      comments.add(title);
     }
 
+    //covert comments to json for proper interpretation in script.js file.
     response.setContentType("application/json;");
-    response.getWriter().println(words);
-  }
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(comments));
+   }
 
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
     String userComment = request.getParameter("user-comments");
-    long timestamp = System.currentTimeMillis();
-
-    //add user comment to ArrayList
-    words.add(userComment);
 
-    //entity with comment property to store user comments
+    //create entity with comment property to store user comments.
     Entity taskEntity = new Entity("Task");
     taskEntity.setProperty("user-comments", userComment);
-    taskEntity.setProperty("timestamp", timestamp);
 
-    //storing using Datastore
+    //add comments to the Datastore for longterm storage. 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(taskEntity);
 
-    // Redirect back to the HTML page.
+    //send the user back to the HTML page after adding a comment.
```

Comment:
"Send the user back to the index page after adding a comment."

Rationale: "the HTML page" might be a little vague.